src/linear_algebra/free_module/strong_rank_condition.lean

```diff
@@ -43,6 +43,10 @@ begin
   { rwa strong_rank_condition_iff_succ R },
   intros n f, by_contradiction hf,
 
+  -- Lean is unable to find this instance without help, either via this `letI`, or via a duplicate
+  -- instance with unecessarily strong typeclasses on `R` and `M`.
+  letI : module.finite R (fin n.succ → R) := module.finite.pi,
```

Comment:
I will let you decide, but I have the impression that it is better to add a global instance that make Lean automatically find this particular one (and maybe others somewhere else).

Reply:
Ah, if you think it will be solved in Lean 4 than OK :)

Reply:
I hope if lean4 doesn't solve it, we can push for it to be changed so that it does :). I'm not opposed to adding the global instance if we keep running into this, but would prefer to wait for more data and keep things conservative in this PR.